### PR TITLE
Pisek Gen V1

### LINKED ISSUE #168: New generator format
Add new generator format with the following convention:
```
./gen
```
Prints all inputs generator can generate.
```
./gen [input_name] [seed]
```
Generates input on given seed.

### PR DESCRIPTION
**Pull request by Tonyl314**
Fixes #168

### CODE REVIEW

File: fixtures/odd_stub/gen.py [hunk 1]
```diff
@@ -4,21 +4,21 @@
 import os
 import sys
 
-random = Random(275)
-directory = sys.argv[1]
+if len(sys.argv) == 1:
+    print("01 repeat=10")
```
Comment (SvizelPritula) [inline fixtures/odd_stub/gen.py:8]: One of the main pros of the new generator format is the ability to have named inputs. The fixtures also serve as examples, so we should encourage good practice and name our inputs. How about `random` of `random-uniform`?

File: pisek/task_jobs/generator/pisek_gen_v1.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
+# pisek  - Tool for developing tasks for programming competitions.
+#
+# Copyright (c)   2024        Antonín Maloň  <git@tonyl.eu>
+# Copyright (c)   2023        Daniel Skýpala <daniel@honza.info>
```
Comment (SvizelPritula) [inline pisek/task_jobs/generator/pisek_gen_v1.py:4]: Did Dan write this last year?

File: pisek/task_jobs/generator/pisek_gen_v1.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    def _complain_about_line(self, line_index: int, contents: str, reason: str) -> None:
+        message = (
+            f"{self.generator}: invalid line {line_index} of listed inputs:\n"
+            f"{contents}\n"
+            f"(reason: {reason})"
+        )
+        raise self._create_program_failure(message, self._run_result)
+
+    def _get_input_lines(self) -> list[str]:
```
Comment (SvizelPritula) [inline pisek/task_jobs/generator/pisek_gen_v1.py:80]: Nit: I wouldn't expect a function named `_get_input_lines` to run the program.

File: pisek/task_jobs/generator/pisek_gen_v1.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    seed: int
+    input_info: InputInfo
+    input_path: TaskPath
+
+    def __init__(self, env: Env, *, name: str = "", **kwargs) -> None:
+        super().__init__(env=env, name=name, **kwargs)
+
+    def _gen(self) -> None:
+        if self.seed < 0:
+            raise ValueError(f"Seed {self.seed} is negative.")
```
Comment (SvizelPritula) [inline pisek/task_jobs/generator/pisek_gen_v1.py:113]: Error messages usually don't start with a capital letter and don't end with a period.

**Review by SvizelPritula — CHANGES_REQUESTED** [review]
There are a couple of small issues.

File: pisek/task_jobs/generator/pisek_gen_v1.py (hunk 4 above)
Comment (Tonyl314) [reply pisek/task_jobs/generator/pisek_gen_v1.py]: I have also fixed this in the opendata_v1 generator.

**Review by SvizelPritula — APPROVED** [review]
LGTM

**Pull request merged**